docs/general/options.md

```diff
@@ -0,0 +1,48 @@
+# Options
+
+## Scriptable Options
+
+Scriptable options also accept a function which is called for each data and that takes the unique argument `context` representing contextual information (see [option context](options.md#option-context)).
+
+Example:
+
+```javascript
+color: function(context) {
+    return context.data < 0 ? 'red' :   // draw negative values in red
+        index%2 ? 'blue' : 'green';    // else, alternate values in blue and green
+}
+```
+
+> **Note:** scriptable options are only supported by a few bubble chart options.
+
+## Indexable Options
+
+Indexable options also accept an array in which each item corresponds to the element at the same index. Note that this method requires to provide as many items as data, so, in most cases, using a [function](#scriptable-options) is more appropriated if supported.
+
+Example:
+
+```javascript
+color: [
+    'red',    // color for data at index 0
+    'blue',   // color for data at index 1
+    'green',  // color for data at index 2
+    'black',  // color for data at index 3
+    //...
+]
+```
+
+## Option Context
+
+The option context is used to give contextual information when resolving options and currently only applies to [scriptable options](#scriptable-options).
+
+The context object contains the following properties:
+
+- `datasetIndex`: index of the current dataset
+- `datasetCount`: number of datasets
+- `dataset`: dataset at index `datasetIndex`
+- `dataIndex`: index of the current data
+- `dataCount`: number of data
+- `data`: value of the current data
+- `chart`: the associated chart
```

Comment:
At least, we need to expose `chart`, `datasetIndex` and `dataIndex` properties.

The other values can be retrieved as follow:
- `datasetCount`: `chart.data.datasets.length`
- `dataset`: `chart.data.datasets[datasetIndex]`
- `dataCount`: `chart.data.datasets[datasetIndex].data.length`
- `data`: `chart.data.datasets[datasetIndex].data[dataIndex]`

`dataset` is always needed to resolve options (since it contains some of them), so it would not cost more to include it to the context. I think most use cases of scriptable options will be to derive the data value, however `data` is (almost) never required to resolve options, so it's an additional cost to expose it to the context. I'm really not sure about exposing (right now) `datasetCount` and `dataCount`.

If we expose only `chart`, `datasetIndex`, `dataIndex` and `dataset`:
- `datasetCount`: `chart.data.datasets.length`
- `dataCount`: `dataset.data.length`
- `data`: `dataset.data[dataIndex]`